src/ncp/ncp_base.cpp

```diff
@@ -3494,13 +3569,40 @@ ThreadError NcpBase::SetPropertyHandler_MAC_SCAN_STATE(uint8_t header, spinel_pr
             break;
 
         case SPINEL_SCAN_STATE_ENERGY:
-            errorCode = otEnergyScan(
-                mInstance,
-                mChannelMask,
-                mScanPeriod,
-                &HandleEnergyScanResult_Jump,
-                this
-            );
+#if OPENTHREAD_ENABLE_RAW_LINK_API
+            if (otLinkRawIsEnabled(mInstance))
+            {
+                // Make sure we aren't already scanning and that we have
+                // only 1 bit set for the channel mask.
+                if (mCurScanChannel == NCP_INVALID_SCAN_CHANNEL &&
+                    HasOnly1BitSet(mChannelMask))
```

Comment:
WHOOPS... mCurScanChannel is what defines the scan state!

I'd just change this check to differentiate between the two error conditions so that they return different error codes.